morse.js

```diff
@@ -62,14 +59,14 @@ window.Morse = (function () {
     return {
         encode: function (text) {
             return text.toUpperCase().split('')
-                .filter(letter => source.hasOwnProperty(letter))
+                .filter(letter => letter in source)
                 .map(letter => letterToMarkup(source[letter]).join(markup.space))
                 .join(byLetter())
         },
         decode: function (source) {
             return source
                 .split(byWord())
-                .map(word => word.split(byLetter()).filter(Boolean).map(letterFromMarkup).join(''))
+                .map(word => word.split(byLetter()).filter(String).map(letterFromMarkup).join(''))
```

Comment:
`Boolean( word.split(byLetter() ))` will always be `true`, because `split` returns an array.

However, if `word` is empty string, `filter(String)` will filter it out
In other words, `filter(Boolean)` is useless here.